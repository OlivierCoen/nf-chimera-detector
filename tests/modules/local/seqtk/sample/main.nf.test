nextflow_process {

    name "Test Process SEQTK_SAMPLE"
    script "modules/local/seqtk/sample/main.nf"
    process "SEQTK_SAMPLE"
    tag "seqtk_sample"

    test("Should run without failures") {

        when {

            process {
                """
                input[0] = [
                    [ sra_id: "test" ],
                    file( '/home/olivier/repositories/nf-chimera-detector/tests/test_data/reads/DRR618977_*.head.fastq.gz', checkIfExists: true),
                    0.1
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
