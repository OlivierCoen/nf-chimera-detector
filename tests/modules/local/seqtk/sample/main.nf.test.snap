{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sra_id": "test"
                        },
                        [
                            "DRR618977_1.head.sampled.fastq.gz:md5,b9eba741a4cceb44fbc62f534f7f4058",
                            "DRR618977_2.head.sampled.fastq.gz:md5,88466f4c0f4c3467756ce9ad8d931e4d"
                        ]
                    ]
                ],
                "1": [
                    [
                        "SEQTK_SAMPLE",
                        "seqtk",
                        "Usage:\n1.1-r93-dirty\n\nCommand:"
                    ]
                ],
                "2": [
                    [
                        "SEQTK_SAMPLE",
                        "pigz",
                        "2.8"
                    ]
                ],
                "reads": [
                    [
                        {
                            "sra_id": "test"
                        },
                        [
                            "DRR618977_1.head.sampled.fastq.gz:md5,b9eba741a4cceb44fbc62f534f7f4058",
                            "DRR618977_2.head.sampled.fastq.gz:md5,88466f4c0f4c3467756ce9ad8d931e4d"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-25T11:57:02.40171326"
    }
}